Wabbajack.App.Blazor/Pages/Install/Configure.razor.cs

```diff
@@ -0,0 +1,131 @@
+using Microsoft.JSInterop;
+using Wabbajack.App.Blazor.State;
+
+namespace Wabbajack.App.Blazor.Pages;
+
+public partial class Configure
+{
+    [Inject] private ILogger<Configure> Logger { get; set; } = default!;
+    [Inject] private IStateContainer StateContainer { get; set; } = default!;
+    [Inject] private DTOSerializer DTOs { get; set; } = default!;
+    [Inject] private SettingsManager SettingsManager { get; set; } = default!;
+    [Inject] private NavigationManager NavigationManager { get; set; } = default!;
+    [Inject] private IJSRuntime JSRuntime { get; set; } = default!;
+    [Inject] private IToastService toastService { get; set; }
+
+    private ModList? Modlist => StateContainer.Modlist;
+    private string ModlistImage => StateContainer.ModlistImage;
+    private AbsolutePath ModlistPath => StateContainer.ModlistPath;
+    private AbsolutePath InstallPath => StateContainer.InstallPath;
+    private AbsolutePath DownloadPath => StateContainer.DownloadPath;
+
+    private InstallState InstallState => StateContainer.InstallState;
+
+    private const string InstallSettingsPrefix = "install-settings-";
+
+    private bool _shouldRender;
+    protected override bool ShouldRender() => _shouldRender;
+
+    protected override async Task OnInitializedAsync()
+    {
+        await LoadModlist();
+        _shouldRender = true;
+    }
+
+    private async Task LoadModlist()
+    {
+        try
+        {
+            if (ModlistPath == AbsolutePath.Empty) throw new FileNotFoundException("Modlist path was empty.");
+            var modlist = await StandardInstaller.LoadFromFile(DTOs, ModlistPath);
+            StateContainer.Modlist = modlist;
+        }
+        catch (Exception e)
+        {
+            toastService.ShowError("Could not load modlist!");
+            Logger.LogError(e, "Exception loading Modlist file {Name}", ModlistPath);
+            NavigationManager.NavigateTo(Select.Route);
+            return;
+        }
+
+        try
+        {
+            var hex = (await ModlistPath.ToString().Hash()).ToHex();
+            var prevSettings = await SettingsManager.Load<SavedInstallSettings>(InstallSettingsPrefix + hex);
+            if (prevSettings.ModlistLocation == ModlistPath)
+            {
+                StateContainer.ModlistPath = prevSettings.ModlistLocation;
+                StateContainer.InstallPath = prevSettings.InstallLocation;
+                StateContainer.DownloadPath = prevSettings.DownloadLocation;
+            }
+        }
+        catch (Exception e)
+        {
+            Logger.LogWarning(e, "Exception loading previous settings for {Name}", ModlistPath);
+        }
+
+        try
+        {
+            var imageStream = await StandardInstaller.ModListImageStream(ModlistPath);
+            var dotnetImageStream = new DotNetStreamReference(imageStream);
+            StateContainer.ModlistImage = new string(await JSRuntime.InvokeAsync<string>("getBlobUrlFromStream", dotnetImageStream));
```

Comment:
why the `new string`? `JSRuntime.InvokeAsync<string>` should return `ValueTask<string>`.

Reply:
I can't remember, fixed now.